src/TMTProductizer/Services/AWS/S3BucketCache.cs

```diff
@@ -0,0 +1,97 @@
+using Amazon.S3;
+using Amazon.S3.Model;
+using System.Text;
+using TMTProductizer.Models;
+using TMTProductizer.Utils;
+namespace TMTProductizer.Services.AWS;
+
+public class S3BucketCache : IS3BucketCache
+{
+    private IAmazonS3 _s3client { get; set; }
+
+    private string _bucketName;
+    private ILogger<S3BucketCache> _logger;
+
+    public S3BucketCache(string bucketName, ILogger<S3BucketCache> logger)
+    {
+        _bucketName = bucketName;
+        _logger = logger;
+        _s3client = new AmazonS3Client();
+    }
+
+    /// <summary>
+    /// Search cache by key, filter by TTL.
+    /// </summary>
+    public async Task<T?> GetCacheItem<T>(string cacheKey)
+    {
+        var typedCacheKey = StringUtils.GetTypedCacheKey<T>(cacheKey);
+
+        // Create a GetObject request
+        var request = new GetObjectRequest
+        {
+            BucketName = _bucketName,
+            Key = $"{typedCacheKey}.json",
+        };
+
+        // Issue request and remember to dispose of the response
+        try
+        {
+            using (GetObjectResponse response = await _s3client.GetObjectAsync(request))
+            {
+                if (response.HttpStatusCode != System.Net.HttpStatusCode.OK)
+                {
+                    _logger.LogWarning($"Bad response for cache key: {typedCacheKey}");
```

Comment:
Kannattaa ehkä logata strign missä on muuttujia näin:

`_logger.LogWarning("Bad response for cache key: {TypedCacheKey}", typedCacheKey);`

Koska:
_"string interpolation prevents structured logging. When you pass the variables after the message, you make it possible for the logger to save them separately. If you just save the log to a file you may not see a difference, but if you later decide to log to a database or in some JSON format, you can just change your logging sink and you will be able to search through the logs much easier without changing all the log statements in your code."_
https://stackoverflow.com/questions/65874828/message-template-should-be-compile-time-constant/65938575#65938575